src/test/components/ProfileCallTreeView.test.js

```diff
@@ -52,17 +52,31 @@ describe('calltree/ProfileCallTreeView', function() {
     expect(calltree.toJSON()).toMatchSnapshot();
   });
 
-  it('renders call tree with a search string', () => {
+  it('renders call tree with some search strings', () => {
     const store = storeWithProfile(profile);
-    store.dispatch(changeCallTreeSearchString('H'));
     const calltree = renderer.create(
       <Provider store={store}>
         <ProfileCallTreeView />
       </Provider>,
       { createNodeMock }
     );
 
-    expect(calltree.toJSON()).toMatchSnapshot();
+    expect(calltree).toMatchSnapshot();
+
+    store.dispatch(changeCallTreeSearchString('C'));
+    expect(calltree).toMatchSnapshot();
```

Comment:
I think testing the search strings using component snapshots will be hard to maintain, as when they break it won't give a useful tip as to why. What do you think about something like the following?

```js
const { profile } = getProfileFromTextSamples(`
  A A A
  B B B
  C C H
  D F I
    E G
    E E
`);

function getHumanReadableSamples (thread: Thread): Array<string | null> {
  // TODO  
}
 
it('can filter threads by search strings', () => {
  const store = storeWithProfile(profile);

  store.dispatch(changeCallTreeSearchString('C'));
  expect(getHumanReadableSamples(getSelectedThread(store.getProfile()))).toEqual([
    'A B C D',
    'A B C F E E',
    null
  ])

  store.dispatch(changeCallTreeSearchString('C, F'));
  expect(getHumanReadableSamples(getSelectedThread(store.getProfile()))).toEqual([
    null,
    'A B C F E E',
    null
  ]);
});
```
Although maybe more granular BDD-style tests.

Reply:
> I think testing the search strings using component snapshots will be hard to maintain, as when they break it won't give a useful tip as to why. What do you think about something like the following?

Actually I found them quite easy to work with. When I accidentally implemented "or" instead of "and", I found it quickly, and also found the reason easily.

Reply:
I don't really want to take the time to implement getHumanReadableSamples so I went forward without this. I looked at it a little bit and it's just more work I'm willing to do at the moment about it.